src/common/connection.ts

```diff
@@ -44,28 +46,24 @@ class Connection extends EventEmitter {
   private _onUnexpectedCloseBound: null|((...args: any[]) => void) = null
   private _fee_base: null|number = null
   private _fee_ref: null|number = null
-  private _connectionTimeout: number
 
   private _trace: (id: string, msg: string) => void = () => {}
+  private _config: ConnectionOptions
 
-  constructor(url, options: ConnectionOptions = {}) {
+  constructor(url?: string, options: ConnectionUserOptions = {}) {
```

Comment:
Why change `url` to optional?

Reply:
Thanks for pointing this out. `url` was untyped before, so it hasn't been changed from required to optional, just from untyped to typed.

As for why it's optional, I was confused by this too. But apparently this is documented behavior to get access to some of the helper methods / signing methods that don't necessarily need a URL to work. (It would be nice to revisit this: what methods are those, and can we pull them off of the Connection object and onto RippleAPI or even just turn them into stateless functions?)

Reply:
Makes sense, thanks!